### ISSUE TITLE
RGS Games Site Down

### ISSUE THREAD

**Issue by HMVocaloid** [esc0rtd3w/realarcade-wrapper-killer#12]
Is the download server gonna be available again? There's some RGS files that I really want but it's impossible to find.

**Comment by esc0rtd3w** [comment]
i will try getting the rgs files uploaded again.

**Comment by esc0rtd3w** [comment]
which ones were you looking for, out of curiosity

**Comment by HMVocaloid** [comment]
There's a lot of games we're looking for, but my brother specifically wants to get Trivia Machine as it would mean everything to him.

**Comment by esc0rtd3w** [comment]
I have uploaded this RGS file for you. 
https://www.mediafire.com/file/v0cebcxryw81uvy/triviamachine_free.rgs/file
I also tested it with the 6.0.0-pretest1 build and it unpacks and patches properly.

**Comment by TheBowlFoSho** [comment]
I have a dozen games I want, but I don't want to ask for them all at once.
So here's a few games I want: Mah Jong Adventures, Gem Shop, Jig Words, and Flying Leo.

**Comment by esc0rtd3w** [comment]
Here ya go! I will see about uploading all of what i have over the next few days. I will make notes with links in github repo.

https://www.mediafire.com/file/u2ptu0t8iv7cx6e/mahjongadventures_free.rgs/file
https://www.mediafire.com/file/qiq9qdw3o9cat4m/gemshop_free.rgs/file
https://www.mediafire.com/file/t7ile9bmjoq50g0/jigwords_free.rgs/file
https://www.mediafire.com/file/5w32vciw4netf7y/flyingleo_free.rgs/file

**Comment by TheBowlFoSho** [comment]
Thanks.
Actually, just one more request. I'd also like Lifetime RSVP.
And a friend of mine would also like Five Card Deluxe and Toy Golf.
And that'll be all for tonight.

**Comment by esc0rtd3w** [comment]
https://www.mediafire.com/file/4ogh4ofnfh77ngr/lifetimersvp_free.rgs/file
https://www.mediafire.com/file/o7b6lesav79xbrw/fivecarddeluxe_free.rgs/file
https://www.mediafire.com/file/on4ukprhkz4olbp/toygolf_free.rgs/file

**Comment by esc0rtd3w** [comment]
I have started uploading the installers here: https://www.mediafire.com/folder/7zj725eafhdyh/installers

I will post this in the main readme also.

**Comment by 1033Forest** [comment]
For tomorrow, I have some games to request...
Full Circle, Glyph 2, Maui Wowee, Mariposa

**Comment by esc0rtd3w** [comment]
all installers will be uploaded here: https://www.mediafire.com/folder/7zj725eafhdyh/installers

Please check there periodically for updated files. I will make a mention when all are complete and add link to main readme.

**Comment by TheBowlFoSho** [comment]
Might you also consider uploading mezzanine files if available?

**Comment by andutrache** [comment]
Thank you so much for these, now I can revisit all my childhood games :)

**Comment by TheBowlFoSho** [comment]
The colletion of installers stopped updating.

**Comment by esc0rtd3w** [comment]
> The colletion of installers stopped updating.

my internet sucks, so i am uploading them all as fast as i can. :)

**Comment by TheBowlFoSho** [comment]
Might it still be possible to get to our requests? The archive seems to be prioritzing Deustche versions of the games over everything else.

**Comment by esc0rtd3w** [comment]
> Might it still be possible to get to our requests? The archive seems to be prioritzing Deustche versions of the games over everything else.

i started with mulilanguage and misc stuff. I will add the other requests after DE versions, which will be soon

**Comment by TheBowlFoSho** [comment]
Okay, thanks for the update. Please take your time.
Whenever available, I would like to request Charm Tale and Word Harmony.

**Comment by 1033Forest** [comment]
Yes, take your time. My next request after the 4 I requested above will be 4 word games.
Academy of Magic - Word Spells, Boggle Supreme, Icy Spell and Word Jolt.

**Comment by esc0rtd3w** [comment]
I dont seem to have these rgs files

Word Spells
Boggle Supreme

The others have been uploaded

**Comment by esc0rtd3w** [comment]
if anyone else has links to missing rgs files (i have a list i will upload of ones i am missing, that i know of) feel free to post them and i can add them to the repository

**Comment by 1033Forest** [comment]
Two games starting with Z next. Zulu Gems and Zzed

**Comment by clydetm** [comment]
> if anyone else has links to missing rgs files (i have a list i will upload of ones i am missing, that i know of) feel free to post them and i can add them to the repository

RealArcade Games Preservation Project collects all .rgs files that can be found to upload here: https://archive.org/details/realrcade-games-preservation-project

**Comment by TheBowlFoSho** [comment]
> I dont seem to have these rgs files
> 
> Word Spells Boggle Supreme
> 
> The others have been uploaded

Word Spells is part of the first game's title. It's not a separate game.
Also, I'm still patiently waiting for Charm Tale and Word Harmony.
Additionally, I would also like to add Shroomz, Holiday Express, and Atlantis (the one RealArcade misspells as "atlants")

**Comment by esc0rtd3w** [comment]
> > if anyone else has links to missing rgs files (i have a list i will upload of ones i am missing, that i know of) feel free to post them and i can add them to the repository
> 
> RealArcade Games Preservation Project collects all .rgs files that can be found and stores them here: https://archive.org/details/realrcade-games-preservation-project Whatever you're missing is either in that already or is sadly lost as far as we know

i have checked these and there are a few there i am missing, which i will add, but i did not also see Boggle

**Comment by esc0rtd3w** [comment]
here is a list of current installers that I have. I will be adding a few missing ones from the RealArcade Games Preservation Project, as stated above, but missing ones may be found on archive.org or similar archiving sites, sitting on old pc's, or are lost forever.

https://github.com/esc0rtd3w/realarcade-wrapper-killer/blob/master/extras/rgs/rgs-archived-installers-list.txt

**Comment by imperialsushi** [comment]
> here is a list of current installers that I have. I will be adding a few missing ones from the RealArcade Games Preservation Project, as stated above, but missing ones may be found on archive.org or similar archiving sites, sitting on old pc's, or are lost forever.
> 
> https://github.com/esc0rtd3w/realarcade-wrapper-killer/blob/master/extras/rgs/rgs-archived-installers-list.txt

thank you kind sir, to say the preservation project is eating good is putting it lightly

**Comment by TheBowlFoSho** [comment]
> here is a list of current installers that I have. I will be adding a few missing ones from the RealArcade Games Preservation Project, as stated above, but missing ones may be found on archive.org or similar archiving sites, sitting on old pc's, or are lost forever.
> 
> https://github.com/esc0rtd3w/realarcade-wrapper-killer/blob/master/extras/rgs/rgs-archived-installers-list.txt

Thank you for the list. Fortunately, it seems all my requests are included in here.
Seeing that it is available, I would also like to request The Office. Please add these six games whenever possible.

**Comment by 1033Forest** [comment]
Please add the two Z games I requested whenever possible. 

Also: Cosmic Bugs, Super Collapse Puzzle Gallery 4, and Tropical Swaps 2.

**Comment by TheBowlFoSho** [comment]
Also, just curious, do you have anything in the way of .mez files?

**Comment by esc0rtd3w** [comment]
> Also, just curious, do you have anything in the way of .mez files?

i do have some. I will try uploading those tonight

**Comment by esc0rtd3w** [comment]
here is also a list of known missing/obsolete RGS installers

https://github.com/esc0rtd3w/realarcade-wrapper-killer/blob/master/extras/rgs/rgs-archived-installers-list-missing.txt

**Comment by TheBowlFoSho** [comment]
Please let me know when my requests are uploaded.

**Comment by esc0rtd3w** [comment]
Requested RGS files have been uploaded. I will upload MEZ files, but i have to switch to hard drive with them on it. I am continuing to upload the rest of the installers, then will do old GH EXE, RGA, and RFS installers.

**Comment by TheBowlFoSho** [comment]
Think I need a break from archiving all the RGSes for now. Might I get a possible ETA on the .mez files?

**Comment by esc0rtd3w** [comment]
> Think I need a break from archiving all the RGSes for now. Might I get a possible ETA on the .mez files?

lol, ive been doing that now off and on for years. I will try and get them uploaded tonight. Ill just add a directory to the rgs folder root, named "mez".

**Comment by TheBowlFoSho** [comment]
In addition to those files, anything else you might have such as .rgp files or client installers may be helpful as well.
I do believe archiving the other aspects of RealArcade are just as important as the games themselves.

**Comment by TheBowlFoSho** [comment]
I see all the RGSes have been uploaded. Thank you for your contributions. I'll eventually add them all to my archive, though for now, I'll upload specific ones.
Will you be able to do the .mez files now?

**Comment by esc0rtd3w** [comment]
> I see all the RGSes have been uploaded. Thank you for your contributions. I'll eventually add them all to my archive, though for now, I'll upload specific ones. Will you be able to do the .mez files now?

the rgs are not finished yet. I will upload mez soon. I promise lol

**Comment by esc0rtd3w** [comment]
> In addition to those files, anything else you might have such as .rgp files or client installers may be helpful as well. I do believe archiving the other aspects of RealArcade are just as important as the games themselves.

you can check the "extras" folder in main repo for some rgp, i believe, and some other misc files. I will also upload any others i find on my other hard drive backups, when i switch from rgs

**Comment by esc0rtd3w** [comment]
> I see all the RGSes have been uploaded. Thank you for your contributions. I'll eventually add them all to my archive, though for now, I'll upload specific ones. Will you be able to do the .mez files now?

i forgot that some mez files are also in the origin dump

Free: https://github.com/esc0rtd3w/realarcade-wrapper-killer/tree/master/extras/dumps/sites/origin.gamehouse.com/gameconsole/games/demo/mezzanines

Full: https://github.com/esc0rtd3w/realarcade-wrapper-killer/tree/master/extras/dumps/sites/origin.gamehouse.com/gameconsole/games/full/mezzanines

I also made zips of both free and full from origin dump and uploaded to mez folder on mediafire.

Free: https://www.mediafire.com/file/43q3z8nywkjuiid/mezzanines-free-20230215.zip/file
Full: https://www.mediafire.com/file/c7g0kuci5udpfsm/mezzanines-full-20230215.zip/file

**Comment by TheBowlFoSho** [comment]
Thank you! Now I can fill in the gape for the metadata in some of the games I downloaded.

**Comment by TheBowlFoSho** [comment]
I mainly ask about the RGP and client files because many of the .rgp files I found in the extras didn't seem to work. (Aside from the Kayak Extreme one), so I was wondering if there was a separate archive for those so I can figure out what they're supposed to do.

As for the clients... we seem to be missing a few versions. I have one version from March 2005, and another one from 2010, but there seems to be a huge gap in between those two.

**Comment by TheBowlFoSho** [comment]
Also, because I noticed some games didn't get uploaded, may I request Thomas and the Magical Words?

**Comment by esc0rtd3w** [comment]
i just recently downloaded 72 RGS files that i was missing (from RealArcade Games Preservation Project). I will add them to the mediafire folder. I will update the lists also.

**Comment by esc0rtd3w** [comment]
> Also, because I noticed some games didn't get uploaded, may I request Thomas and the Magical Words?

dude, they are still uploading lol. Its around 100gb and i have crap internet. I said i would post here when they are finished. Ill add that one after the D's are finished. I added the smaller ones first, as thats how mediafire does it by default anyways.

**Comment by esc0rtd3w** [comment]
> Also, because I noticed some games didn't get uploaded, may I request Thomas and the Magical Words?

i just uploaded that one for you

https://www.mediafire.com/file/dq07hcbhko0pkjd/thomasandthemagicalwords_free.rgs/file

**Comment by andutrache** [comment]
> > Also, because I noticed some games didn't get uploaded, may I request Thomas and the Magical Words?
> 
> dude, they are still uploading lol. Its around 100gb and i have crap internet. I said i would post here when they are finished. Ill add that one after the D's are finished. I added the smaller ones first, as thats how mediafire does it by default anyways.

Forgive us for being impatient, its basically because we are excited when we get to see our childhood games again.
Also wow 100 GB, my HDD is crying but the only tears I am shedding are tears of joy :)

**Comment by TheBowlFoSho** [comment]
> > > Also, because I noticed some games didn't get uploaded, may I request Thomas and the Magical Words?
> > 
> > 
> > dude, they are still uploading lol. Its around 100gb and i have crap internet. I said i would post here when they are finished. Ill add that one after the D's are finished. I added the smaller ones first, as thats how mediafire does it by default anyways.
> 
> Forgive us for being impatient, its basically because we are excited when we get to see our childhood games again. Also wow 100 GB, my HDD is crying but the only tears I am shedding are tears of joy :)

Yeah, pretty much that. It took me so long to find and be able to play these games, so my apologies if I come across as impatient.

**Comment by esc0rtd3w** [comment]
Finally, all the RGS installers that I have are uploaded.

**Comment by andutrache** [comment]
All 138 GB of RGSes have been downloaded. Thanks <3

**Comment by HMVocaloid** [comment]
Now that we've got mostly all of the games archived and we're working on creating working RGP files, would you consider joining our server to help with reverse engineering? We want to get the download button functional.

**Comment by TheBowlFoSho** [comment]
We've been successful with making RGP files and remapping them to our archive.
However... is there a way to reserve engineer RealArcade so that they can be downloaed from there?
Like, pretty much what he said, remapping the download button to somewhere else.

**Comment by esc0rtd3w** [comment]
> Now that we've got mostly all of the games archived and we're working on creating working RGP files, would you consider joining our server to help with reverse engineering? We want to get the download button functional.

hmmm, ok. Interesting. I dont know much about the RGP file format. When you say remapping the download button, what do you mean? Using the original RNArcade software? I could probably join the server at some point. Is it discord, because i have to make a new account if so.

**Comment by esc0rtd3w** [comment]
> Now that we've got mostly all of the games archived and we're working on creating working RGP files, would you consider joining our server to help with reverse engineering? We want to get the download button functional.

hmmm, ok. Interesting. I dont know much about the RGP file format. When you say remapping the download buttone, what do you mean? Using the original RNArcade software? I could probably join the server at some point. Is it discord, becasue i have to make a new account if so.

**Comment by esc0rtd3w** [comment]
> We've been successful with making RGP files and remapping them to our archive. However... is there a way to reserve engineer RealArcade so that they can be downloaded from there? Like, pretty much what he said, remapping the download button to somewhere else.

Ok, so i assume the RGP files are like metadata for file info, url, etc? I need to look further into that because i havent really messed with making custom ones, or even looking into the format too much. I modified the RNArcade software a while ago and embedded HTML links into it and added new buttons. All of my work should be in this GitHub repo, as far as I know. I have been adding missing stuff from my backups. That is also how the RGS Loader works, by using an old beta version of the RNArcade exe, modified to just extract the RGS files, using same method as it does for catalog updates.

It is quite interesting that others are also investing time into reversing RNA stuff. I thought i was the only one for a long time lol.

I did most of my reversing of RNA and game patching using OllyDBG, but recently over the past few years have been using IDA for PS3 stuff and random PC stuff, and wouldnt mind loading it into there and taking a look. I will try to continue adding whatever i find to this repo, but i will also try helping others with similar goals, and the RGP stuff is intriguing me now lol

I also want to do a bit more archive hunting for lost files when i have free time, as we are still missing hundreds, if not thousands of RGS installers and other formats like RGA and RFS. The latter being more available in most cases, but not released older games in these new formats.

**Comment by TheBowlFoSho** [comment]
> > Now that we've got mostly all of the games archived and we're working on creating working RGP files, would you consider joining our server to help with reverse engineering? We want to get the download button functional.
> 
> hmmm, ok. Interesting. I dont know much about the RGP file format. When you say remapping the download button, what do you mean? Using the original RNArcade software? I could probably join the server at some point. Is it discord, because i have to make a new account if so.

So, I found out the RGP files try to download from  the RealArcade site, and because it no longer exists, it doesn't work. So I made a separate archive to act as a server for it to download from (it downloads the RGS installer, mezzanine, and the cabinet, or "buy now" advertisement, if the game is older than 2004), and the RGP downloaded all three files successfully. 
However, there does not seem to be any way to download the RGP directly from RealArcade as there was back then.

Also, yes, it is on Discord.

**Comment by esc0rtd3w** [comment]
> > > Now that we've got mostly all of the games archived and we're working on creating working RGP files, would you consider joining our server to help with reverse engineering? We want to get the download button functional.
> > 
> > 
> > hmmm, ok. Interesting. I dont know much about the RGP file format. When you say remapping the download button, what do you mean? Using the original RNArcade software? I could probably join the server at some point. Is it discord, because i have to make a new account if so.
> 
> So, I found out the RGP files try to download from the RealArcade site, and because it no longer exists, it doesn't work. So I made a separate archive to act as a server for it to download from (it downloads the RGS installer, mezzanine, and the cabinet, or "buy now" advertisement, if the game is older than 2004), and the RGP downloaded all three files successfully. However, there does not seem to be any way to download the RGP directly from RealArcade as there was back then.
> 
> Also, yes, it is on Discord.

quite honestly, fuck those ads lol.

So basically we would need to recreate the same structure that RA had at that time. This is less interesting to me than archiving the lost files and getting the games to unpack and play, but i do see the interest in what you are doing, and it all helps the same cause. I should have time maybe this weekend and i can join the server and toss around a few ideas.

**Comment by TheBowlFoSho** [comment]
> 
> quite honestly, fuck those ads lol.
> 

Eh, I just keep them there because I find them nostalgic lol
But also for preservation purposes.

**Comment by HMVocaloid** [comment]
I know this is pretty unlikely, but do you happen to have any of the Realarcade server files? I'm trying to reconstruct the server directory for the latest version.

**Comment by TheBowlFoSho** [comment]
This may be highly unlikely , but would you happen to have an archived collection of every game's wrapper.cab as well as the pngs that appear asking to buy the game?
Now yes, I know the whole intent of the wrapper killer is to get rid of these things lol.  However, as me and my brother are making a total reconstruction of RealArcade's server, I would like to make sure I have the complete metadata for every game possible, which would include the cabinets/advertisements. What anyone else does with them is really their prerogative, but I personally would like to have them all.

Update: I have found that many of them can be easily recreated with very little issue. However, several games (such as Mah Jong Adventures, Big Kahuna Reef, Sally's Salon, and Charm Solitaire, to name a few) don't seem to have any artwork. So if you have them, I'd like them for the cabs.
